ui/main.cpp

```diff
@@ -172,6 +197,101 @@ class CrashHandler
     }
 };
 
+//--------------------------------------------------------------------------------------------------
+class LogHandler
+{
+public:
+    static void Initialize(const char *argv0)
+    {
+        // Use a fixed, non-timestamped filename for the standard application log
+        QString filename = "dive" + QDateTime::currentDateTime().toString("yyyyMMdd-HHmmss") +
+                           ".log.txt";
+        LogPaths paths = ResolveBaseLogPaths(argv0, filename);
+
+        // Test for writability to choose the final path
+        QFile test_file(paths.primary_path);
+
+        if (test_file.open(QIODevice::WriteOnly | QIODevice::Append))
+        {
+            m_resolved_path = paths.primary_path;
+            test_file.close();
+        }
+        else
+        {
+            m_resolved_path = paths.fallback_path;
+        }
+
+        // Install the static message handler function
+        qInstallMessageHandler(LogHandler::MessageOutput);
+
+        qDebug() << "Application log handler initialized. Log file path:" << m_resolved_path;
+    }
+
+private:
+    inline static QString m_resolved_path;
+    inline static QMutex  m_mutex;
+
+    static void MessageOutput(QtMsgType type, const QMessageLogContext &context, const QString &msg)
+    {
+        QMutexLocker lock(&m_mutex);
+
+        QString log_level_prefix;
+        switch (type)
+        {
+        case QtDebugMsg:
+            log_level_prefix = "DEBUG";
+            break;
+        case QtInfoMsg:
+            log_level_prefix = "INFO";
+            break;
+        case QtWarningMsg:
+            log_level_prefix = "WARNING";
+            break;
+        case QtCriticalMsg:
+            log_level_prefix = "CRITICAL";
+            break;
+        case QtFatalMsg:
+            log_level_prefix = "FATAL";
+            break;
+        default:
+            log_level_prefix = "UNKNOWN";
+            break;
+        }
+
+        // Format the log line: [Time] [Level] Message
+        QString currentTime = QDateTime::currentDateTime().toString("yyyy-MM-dd hh:mm:ss.zzz");
+        QString formatted_message = QString("[%1] [%2] %3 (%4:%5 %6)")
+                                    .arg(currentTime)
+                                    .arg(log_level_prefix)
+                                    .arg(msg)
+                                    .arg(QFileInfo(context.file).fileName())
+                                    .arg(context.line)
+                                    .arg(context.function);
+
+        // Output to console only if the flag is set
+        if (absl::GetFlag(FLAGS_log_to_console))
+        {
+            fprintf(stderr, "%s\n", formatted_message.toLocal8Bit().constData());
+        }
+
+        // Write to file using the resolved path
+        QFile logFile(m_resolved_path);
+
+        if (logFile.open(QIODevice::WriteOnly | QIODevice::Append))
```

Comment:
You probably want a file that is opened for the entire lifetime of the application.